packages/connect-core/src/utils/misc.ts

```diff
@@ -1,3 +1,54 @@
 export function toArrayEntry<T>(value: T | T[]): T[] {
   return Array.isArray(value) ? value : [value]
 }
+
+type CacheStoreCallback<T> = () => Promise<T>
+type CacheStoreCached<T> = (
+  id: string,
+  callback: CacheStoreCallback<T>
+) => Promise<T>
+
+export function createCacheStore<T = string>(
+  limit: number = 10
+): {
+  cachedIndex: (id: string) => number
+  clear: () => void
+  get: (id: string, callback: CacheStoreCallback<T>) => Promise<T>
+  touch: (index: number) => void
+} {
+  const cachedEntries: [string, T][] = Array(limit)
+
+  return {
+    clear(): void {
+      cachedEntries.length = 0
+    },
+    cachedIndex(id: string): number {
+      return cachedEntries.findIndex((entry) => entry && id === entry[0])
+    },
+    touch(index: number): void {
+      cachedEntries.unshift(cachedEntries.splice(index, 1)[0])
```

Comment:
Here we are using `splice` because we want to mutate the cache and remove the element from `index` and place it first in the array?

Reply:
Exactly yes, just a trick to move `index` at the first position :-)